File: Src/LibNoise/Generators/Spheres.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace LibNoise
{
    public class Spheres
        : IModule
    {
        public double Frequency { get; set; }

        public Spheres()
        {
            Frequency = 1.0;
        }

        public double GetValue(double x, double y, double z)
        {
            x *= Frequency;
            y *= Frequency;
            z *= Frequency;

            double distFromCenter = System.Math.Sqrt(x * x + y * y + z * z);
            int distFromCenter0 = (distFromCenter > 0.0 ? (int)distFromCenter : (int)distFromCenter - 1);
//            int distFromCenter0 = (x > 0.0 ? (int)x : (int)x - 1);
            double distFromSmallerSphere = distFromCenter - distFromCenter0;
            double distFromLargerSphere = 1.0 - distFromSmallerSphere;
            double nearestDist = NMath.GetSmaller(distFromSmallerSphere, distFromLargerSphere);
            return 1.0 - (nearestDist * 4.0); // Puts it in the -1.0 to +1.0 range.
        }
    }
}

File: js/NoiseLib/Generators/Spheres.js
LibNoise.Spheres = function()
{
  this.Frequency = 1.0;
  this.Name = "LibNoise.Spheres";
}

LibNoise.Spheres.prototype.getInput = getNone;
LibNoise.Spheres.prototype.setInput = setNone;

LibNoise.Spheres.prototype.GetValue = function (x, y, z)
{
  x *= this.Frequency;
  y *= this.Frequency;
  z *= this.Frequency;

  var distFromCenter = Math.sqrt(x * x + y * y + z * z);
  var distFromCenter0 = Math.floor(distFromCenter > 0.0 ? distFromCenter : distFromCenter - 1);
  var distFromSmallerSphere = distFromCenter - distFromCenter0;
  var distFromLargerSphere = 1.0 - distFromSmallerSphere;
  var nearestDist = LibNoise.NMath.GetSmaller(distFromSmallerSphere, distFromLargerSphere);
  return 1.0 - (nearestDist * 4.0);
}
